File: s3GoSDKExample/sample/Cleanup.go
package main

import (
	"context"
	"errors"
	"fmt"

	"github.com/aws/aws-sdk-go-v2/service/s3"
	"github.com/aws/smithy-go"

)

func CleanUp(buckets [3]string) {
	for _, bucket := range buckets {
		input2 := &s3.ListObjectsInput{
			Bucket: &bucket,
		}
		obj, listobjectsErr := Client.ListObjects(context.TODO(), input2)
		if listobjectsErr != nil {
			fmt.Printf("Could not list object in " + bucket +": ")
			var ae smithy.APIError
			if errors.As(listobjectsErr, &ae) {
				fmt.Println(ae.ErrorCode())
			}
			continue
		}
		for _, object := range obj.Contents {
			output := &s3.DeleteObjectInput{
				Bucket: &bucket,
				Key:    object.Key,
			}
			_, deleteobjectErr := Client.DeleteObject(context.TODO(), output)
			if deleteobjectErr != nil {
				fmt.Println("Got an error deleting item:")
				fmt.Println(deleteobjectErr)
				continue
			}
			fmt.Println("Deleted " + *object.Key + " from " + bucket)
		}
		input3 := &s3.DeleteBucketInput{
			Bucket: &bucket,
		}
		_, deleteErr := Client.DeleteBucket(context.TODO(), input3)
		if deleteErr != nil {
			fmt.Println("Could not delete bucket " + bucket)
			fmt.Println(deleteErr)
			continue
		}
		fmt.Println("Delete bucket : " + bucket)
	}
	fmt.Println("")
}

File: s3RubySDK/sample/cleanup.rb
def cleanup(s3, buckets)
  puts "===============cleanup==============="
  buckets.each do |bucket|
    resp = s3.list_object_versions({
      bucket: bucket,
    })
    resp.versions.each do |obj|
      puts "delete object: #{bucket}/#{obj.key} #{obj.version_id} "
      s3.delete_object({
        bucket: bucket,
        key: obj.key,
        version_id: obj.version_id,
      })
    end
    resp.delete_markers.each do |marker|
      puts "Rm delete marker: #{bucket}/#{marker.key} #{marker.version_id} "
      s3.delete_object({
        bucket: bucket,
        key: marker.key,
        version_id: marker.version_id,
      })
    end
    puts "delete bucket: #{bucket}"
    s3.delete_bucket({
      bucket: bucket,
    })
    puts "delete success"
  rescue Aws::S3::Errors::NoSuchBucket => e
  end
end
